src/main/scala/mesosphere/mesos/EnvironmentHelper.scala

```diff
@@ -87,10 +87,11 @@ object EnvironmentHelper {
           env += (s"PORT_$generatedPort" -> generatedPort.toString)
       }
 
-      requestedPorts.map(_.name).zip(effectivePorts).foreach {
-        case (Some(portName), Some(effectivePort)) =>
+      requestedPorts.zip(effectivePorts).foreach {
+        case (PortRequest(Some(portName), _), Some(effectivePort)) =>
           env += (s"PORT_${portName.toUpperCase}" -> effectivePort.toString)
-        // TODO(jdef) port name envvars for generated container ports
+        case (PortRequest(Some(portName), port), None) =>
```

Comment:
should consider generated container ports here (e.g. user declared a container port of 0) otherwise you might end up with some envvar PORT_FOO=0

Reply:
if the intent is NOT to generate envvars for generated containers ports, then ... please reinstate the deleted TODO